src/backend/gencomm.py

```diff
@@ -313,10 +319,12 @@ def populate_pupfns(pup_max_nesting, backend, blklens, builtin_types, builtin_ma
     yutils.display(OUTFILE, "#ifndef YAKSURI_%sI_POPULATE_PUPFNS_H_INCLUDED\n" % backend.upper())
     yutils.display(OUTFILE, "#define YAKSURI_%sI_POPULATE_PUPFNS_H_INCLUDED\n" % backend.upper())
     yutils.display(OUTFILE, "\n")
-    for dtype1 in derived_types:
-        for dtype2 in derived_types:
-            yutils.display(OUTFILE, "int yaksuri_%si_populate_pupfns_%s_%s(yaksi_type_s * type);\n" % (backend, dtype1, dtype2))
-        yutils.display(OUTFILE, "int yaksuri_%si_populate_pupfns_%s_builtin(yaksi_type_s * type);\n" % (backend, dtype1))
+    if pup_max_nesting > 0:
+        for dtype1 in derived_types:
+            if pup_max_nesting > 1:
```

Comment:
This makes depth=1 equivalent to depth=0, which can be simplified by cosolidate the condition to outer layer.

How about rename original `populate_pupfns` to `populate_pupfns_level_2` and add `populate_pupfns_level_0` and `populate_pupfns_level_1` and have `populate_pupfns` conditionally call each? `populate_pupfns_level_0` is likely to be empty and can be omitted.

Reply:
I see the depth=1 is being generated. I guess the patch is fine. Sorry for the noise.